### PR TITLE
Various fixes accumulated so far

### PR DESCRIPTION
**Pull request by chanseokoh**
Some are cleanups, but some are crucial.

### CODE REVIEW

**Comment by chanseokoh** [comment]
Ah... wait. I need to back out one change.

File: features/com.google.cloud.tools.eclipse.3rdparty.feature/feature.xml [hunk 1]
```diff
@@ -78,7 +78,7 @@
          id="org.yaml.snakeyaml"
          download-size="0"
          install-size="0"
-         version="1.17"
+         version="1.17.0"
```
Comment (elharo) [inline features/com.google.cloud.tools.eclipse.3rdparty.feature/feature.xml]: I could swear there was a reason for this. Could you pull this file into a separate PR in case we need to roll this bit back? Thanks.

Comment (chanseokoh) [inline features/com.google.cloud.tools.eclipse.3rdparty.feature/feature.xml]: Sure. Done.
